### PR TITLE
Check for new "derivationOrigin" field in id_alias

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Fix issue raised here: https://dfinity.atlassian.net/browse/FOLLOW-1651 which is already merged in II: https://github.com/dfinity/internet-identity/pull/2664

# Changes

* Add parameter `expected_derivation_origin` to `AliasTuple` and to verify functions. Use the new param to check the claims of the id alias credential.

# Tests

There were some hidden dependencies among tests and constants used. For example, the expiration was the same in all the credentials.
* I created new variables and scoped the dependencies within the test if possible.
* I kept some JWS because they are hard to replicate and there was no need to redo again. Those are the ones that use a local IC root key.
* I added new JWS with the new id_alias used for most of the tests.
* I added two new tests. One to check that a wrong derivation origin raises an error and another one that the verification fails with the old JWS.

# Todos

- [ ] Add entry to changelog (if necessary).

### CODE REVIEW

File: rust-packages/ic-verifiable-credentials/src/lib.rs [hunk 1]
```diff
@@ -1360,145 +1421,156 @@ mod tests {
...
+            &mainnet_test_vc_flow_signers(),
+            &vp_vc_spec(),
+            &mainnet_ic_root_pk_raw(),
+            VP_CURRENT_TIME_AFTER_EXPIRY_NS,
         );
         assert_matches!(result, Err(e) if format!("{:?}", e).to_string().contains("credential expired"));
     }
 
     #[test]
-    fn should_fail_validate_ii_presentation_and_claims_if_wrong_vcs() {
```
Comment (lmuntaner) [inline rust-packages/ic-verifiable-credentials/src/lib.rs:1487]: I changed this test to make it more obvious. The only difference before between `VERIFIED_EMPLOYEE_VC_FOR_VP_JWS` and `VERIFIED_ADULT_VC_FOR_VP_JWS` was the credential spec. 

Therefore, the test is the same as pass a wrong spec.

File: rust-packages/ic-verifiable-credentials/src/lib.rs [hunk 2]
```diff
@@ -1360,149 +1421,220 @@ mod tests {
...
+            &mainnet_test_vc_flow_signers(),
+            &wrong_spec,
+            &mainnet_ic_root_pk_raw(),
+            VP_CURRENT_TIME_BEFORE_EXPIRY_NS,
         );
         assert_matches!(result, Err(e) if format!("{:?}", e).to_string().contains("InconsistentCredentialJwtClaims"));
     }
 
+    #[test]
+    fn should_fail_validate_ii_presentation_and_claims_if_different_derivation_origin() {
```
Comment (lmuntaner) [inline rust-packages/ic-verifiable-credentials/src/lib.rs:1579]: This is a new test.

File: rust-packages/ic-verifiable-credentials/src/lib.rs [hunk 3]
```diff
@@ -1360,149 +1421,220 @@ mod tests {
...
+            &mainnet_test_vc_flow_signers(),
+            &vp_vc_spec(),
+            &mainnet_ic_root_pk_raw(),
+            VP_CURRENT_TIME_BEFORE_EXPIRY_NS,
+        );
+        assert_matches!(result, Err(e) if format!("{:?}", e).to_string().contains("unexpected derivation origin"));
+    }
+
+    #[test]
+    fn should_fail_validate_ii_presentation_and_claims_if_missing_derivation_origin() {
```
Comment (lmuntaner) [inline rust-packages/ic-verifiable-credentials/src/lib.rs:1600]: This is a new test.

**Comment by lmuntaner** [comment]
@frederikrothenberger please review

File: Cargo.lock [hunk 4]
```diff
@@ -3368,4 +3368,4 @@ dependencies = [
  "sha3",
  "thiserror",
  "zeroize",
-]
+]
```
Comment (frederikrothenberger) [inline Cargo.lock:3371]: Why do we have this change?

**Review by frederikrothenberger — APPROVED** [review]
Thanks, LGTM!

File: Cargo.lock (hunk 4 above)
Comment (lmuntaner) [reply Cargo.lock:3371]: Oh, I had to revert changes here. I'll add the EOF line

**Pull request merged**